src/dvclive/live.py

```diff
@@ -391,14 +394,20 @@ def log_image(self, name: str, val):
     def log_plot(
         self,
         name: str,
-        datapoints: List[Dict],
+        datapoints: pd.DataFrame | np.ndarray | List[Dict],
         x: str,
         y: str,
         template: Optional[str] = None,
         title: Optional[str] = None,
         x_label: Optional[str] = None,
         y_label: Optional[str] = None,
+        columns: Optional[List[str]] = None,
```

Comment:
I'm not sure we need this. It's only useful for the specific scenario of labeling numpy array columns AFAICT, and there are already several ways to handle that which are at least as straightforward IMO:

1. Convert to a pd dataframe yourself
2. Use [structured arrays](https://numpy.org/doc/stable/user/basics.rec.html)